Tasks/Common/webdeployment-common/deployusingmsdeploy.ts

```diff
@@ -22,28 +23,41 @@ export async function DeployUsingMSDeploy(webDeployPkg, webAppName, publishingPr
         excludeFilesFromAppDataFlag, takeAppOfflineFlag, virtualApplication, setParametersFile, additionalArguments, isFolderBasedDeployment, useWebDeploy) {
 
     setParametersFile = utility.getSetParamFilePath(setParametersFile);
+    var setParametersFileName = null;
+    var pathVar;
+    if(setParametersFile != null) {
+        setParametersFileName = setParametersFile.slice(setParametersFile.lastIndexOf('\\') + 1, setParametersFile.length);
+    }
     var isParamFilePresentInPackage = isFolderBasedDeployment ? false : await msDeployUtility.containsParamFile(webDeployPkg);
     var msDeployPath = await msDeployUtility.getMSDeployFullPath();
     var msDeployCmdArgs = msDeployUtility.getMSDeployCmdArgs(webDeployPkg, webAppName, publishingProfile, removeAdditionalFilesFlag,
-        excludeFilesFromAppDataFlag, takeAppOfflineFlag, virtualApplication, setParametersFile, additionalArguments, isParamFilePresentInPackage, isFolderBasedDeployment, 
+        excludeFilesFromAppDataFlag, takeAppOfflineFlag, virtualApplication, setParametersFileName, additionalArguments, isParamFilePresentInPackage, isFolderBasedDeployment, 
```

Comment:
For web packages - you are creating a copy of param file in working directory. Hence I believe you are only passing the name.
For folder - I don't see you are making any copy. How does that work in that case?